app/services/static_members_recruiter.rb

```diff
@@ -0,0 +1,77 @@
+# frozen_string_literal: true
+
+# It enroll members with static slots to the related missions
+class StaticMembersRecruiter < ApplicationService # rubocop:disable Style/ClassAndModuleChildren
+  def initialize
+    @errors = []
+    @static_slots = StaticSlot.all
+    @members_with_static_slots = Member.all.select { |member| member.static_slots.any? }
+    @current_time = DateTime.current
+  end
+
+  def call
+    enrollments_for_all_members
+  end
+
+  def enrollments_for_all_members
+    @members_with_static_slots.each do |member|
+      enrollment_for_one_member(member)
+    end
+  end
+
+  def enrollment_for_one_member(member)
+    member.static_slots.each do |static_slot|
+      unless already_enrolled_for_this_time_slot?(member, static_slot)
+        enrollments_for_one_static_slot(member, static_slot)
+      end
+    end
+  end
+
+  def enrollments_for_one_static_slot(member, static_slot)
+    mission_slots = search_mission_slots_corresponding_to_static_slot(static_slot)
+    mission_slots.each do |mission_slot|
+      mission_slot.update(member_id: member.id) unless already_enrolled_for_this_time_slot?(member, mission_slot)
+    end
+  end
+
+  def search_mission_slots_corresponding_to_static_slot(static_slot)
+    mission_slots = []
+    start_time = determine_first_start_time_corresponding_to_static_slot_after_current_date(static_slot)
+    mission_slots << Mission::Slot.find_by(start_time: start_time, member_id: nil)
+    mission_slots << Mission::Slot.find_by(start_time: start_time + 4.weeks, member_id: nil)
+    mission_slots.compact
+  end
+
+  def determine_first_start_time_corresponding_to_static_slot_after_current_date(static_slot)
+    first_start_time = @current_time + determine_intervals_with_current_time(static_slot)
+    first_start_time = DateTime.new(first_start_time.year,
+                                    first_start_time.month,
+                                    first_start_time.day,
+                                    static_slot.start_time.hour,
+                                    static_slot.start_time.min)
+    first_start_time += 4.weeks if first_start_time < @current_time
+    first_start_time
+  end
+
+  def determine_intervals_with_current_time(static_slot)
+    gaps = {}
+    type_of_current_week = determine_week_type(@current_time)
+    gaps['week'] = ((StaticSlot.week_types[static_slot.week_type] + 1) - (StaticSlot.week_types[type_of_current_week] + 1 )) % 4
+    gaps['day'] = ((StaticSlot.week_days[static_slot.week_day] + 1) - @current_time.strftime('%u').to_i) % 7
+    gaps['week'] -= 1 if (StaticSlot.week_days[static_slot.week_day] + 1) < @current_time.strftime('%u').to_i
+    (gaps['week'].weeks + gaps['day'].days)
+  end
```

Comment:
Could you put an explanatory comment here please? I have no idea what it does and why